File: app/src/main/java/com/yineng/yishizhizun/common/js/CallJSRunnable.java
package com.yineng.yishizhizun.common.js;

import com.yineng.yishizhizun.common.WebViewManager;

/**
 * Created by "iqos_jay@outlook.com" on 2018/6/27.
 */
public class CallJSRunnable implements Runnable {

    private final String func;

    public CallJSRunnable(String func) {
        this.func = func;
    }

    @Override
    public void run() {
        if (WebViewManager.isX5Core()) {
            com.tencent.smtt.sdk.WebView webView = (com.tencent.smtt.sdk.WebView) JSEnv.getEnv(JSEnv.WEBVIEW);
            webView.evaluateJavascript(func, null);
        } else {
            android.webkit.WebView webView = (android.webkit.WebView) JSEnv.getEnv(JSEnv.WEBVIEW);
            webView.evaluateJavascript(func, null);
        }
    }
}

File: app/src/main/java/com/kuplay/kuplay/common/js/CallJSRunnable.kt
package com.kuplay.kuplay.common.js

import com.kuplay.kuplay.base.BaseWebView
import com.kuplay.kuplay.base.YNWebView
import com.kuplay.kuplay.widget.AndroidWebView
import com.kuplay.kuplay.widget.X5Chrome

/**
 * Created by "iqos_jay@outlook.com" on 2018/6/27.
 */
class CallJSRunnable(var webView: Any, private val func: String) : Runnable {
    override fun run() {
        if (webView == null) webView = JSEnv.getEnv(JSEnv.WEBVIEW) as Object
        YNWebView.evaLua(webView,func)
    }
}
